### PR TITLE
Feature/windowsy linux

### PR DESCRIPTION
**Pull request by nicholas-sly**
Added MinGW and Cygwin options in the CMakeLists.txt generation code.

### CODE REVIEW

File: cmake.py [hunk 1]
```diff
@@ -18,11 +19,18 @@ def fetch_subprojects(state):
...
+                message( FATAL_ERROR "git installation was not found." )
+            endif()
+        endif()
+
         if( NOT ROOT_DIRECTORY )
             set( ROOT_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} )
             if ( NOT fetched_subprojects )
                 if ( NOT PYTHON_EXECUTABLE )
-                    find_package( PythonInterp )
+                    find_package( PythonInterp 3.4 )
```
Comment (jlconlin) [inline cmake.py:33]: I wonder how strict we want to be on the Python version right now. While we do stay that we require Python 3.4, there is nothing we do that requires Python 3.x. @apmccartney @whaeck care to weigh in on this?

File: cmake.py [hunk 2]
```diff
@@ -643,6 +651,9 @@ def generate():
         contents += add_targets(state)
         contents += add_tests(state) 
         contents += install(state)
+        contents += """
```
Comment (jlconlin) [inline cmake.py:654]: 👍

**Review by jlconlin — COMMENTED** [review]
Overall, looks good. I'd like to hear some feedback from @apmccartney before this gets approved as he knows metaconfigure better than I do.

**Comment by nicholas-sly** [comment]
As no one has provided input, can this be approved?  This combines with my pull request to spdlog-adapter to allow NJOY21 to build using Cygwin (with static libraries only).  These two pull requests can get at least one version of NJOY21 on Windows building on the Jenkins testing server.

**Review by jlconlin — APPROVED** [review]

**Pull request merged**